docs/federation.md

```diff
@@ -0,0 +1,126 @@
+  * `GSLBConfig federation`: The federator indicates whether it was able to federate the `GSLBConfig` object to all the clusters in `spec.clusters` successfully.
+  * `GDP Federation`: The federator indicates whether it was able to federate the `GDP`/`GlobalDeploymentPolicy` object to all the clusters in `spec.clusters` successfully.
+
+**Note** that if `helm` is used to deploy AMKO, this Custom Resource will be installed, and the users have to provide these values via `values.yaml`.
+
+
+**Additional Notes**:
+* The federation set can be a subset of the overall member cluster set used for GSLB.
+* AMKO has to be deployed on all clusters in the federation set.
+* `spec.clusterContext` must contain the current cluster's context.
+* `spec.version` is compared against the versions of all member clusters. If there's a version mismatch, the current AMKO will error out and the AMKO federator won't federate.
+* `spec.clusters` contains the federation cluster set.
+* Only one AMKO can be a leader, all other AMKOs have to be followers. If there are two leaders at any point, federation will stop and the error will be written to the `AMKOCluster`'s status.
+
+### Disasters and Recovery
+Whenever a disaster occurs and it takes down the cluster where the leader AMKO is deployed, the kubernetes/openshift objects will not be synced to the Avi GSLB Leader Controller anymore. At this point, the user has to switch one of the follower AMKOs to leader and the syncing should just resume.
+
+Let's assume that there are 3 sites with one cluster in each of them:
+1. Cluster 1 in Site 1
+2. Cluster 2 in Site 2
+3. Cluster 3 in Site 3
+
+A disaster can occur either in the entire site or just for that cluster. Site failure would also mean that the kubernetes/openshift cluster along with the Avi Controller in that site are down. Whereas, a cluster failure would mean that only the kubernetes/openshift cluster is down.
+
+#### Site Failure Scenario
+
+ ![Alt text](images/amko_fed_switch.png?raw=true "AMKO site failure switch")
+
+The site where the leader AMKO was deployed and which hosted the Avi GSLB leader, fails. At this point, the user has to:
+1. Choose a new Avi GSLB leader controller out of the GSLB followers by following [these](https://avinetworks.com/docs/20.1/how-to-select-a-leader-site-manually/) steps.
+2. Choose a new follower AMKO to be the new leader, and perform the following steps on the cluster where this follower AMKO is deployed:
+  a. Edit the `GSLBConfig` object and change the leader IP address:
+
+        $ kubectl edit gslbconfig -n avi-system gc-1
+        // Set the field spec.gslbLeader.controllerIP to the new leader's IP address
+
+  b. Set the `isLeader` field in the `AMKOCluster` object to true on this cluster:
+
+    $ kubectl edit amkocluster amkocluster-federation -n avi-system
+
+This would reboot AMKO in leader mode and will resume syncing objects to the Avi GSLB leader controller.
```

Comment:
This would reboot AMKO in leader mode and will resume syncing objects to the Avi GSLB leader controller.

Consider changing to:

This would reboot the new leader AMKO. Post reboot, the new `leader` will take over the responsibilities of the earlier leader.